### DISCUSSION TITLE
Provide possibility to configure the firebase app purely from Flutter code

### DISCUSSION THREAD

**Discussion by devjeff** [firebase/flutterfire#3926] [Feature request]
With the new version of firebase_core it is already possible to configure the firebase app instance directly from Flutter code by passing the firebase options to the initializeApp method (https://firebase.flutter.dev/docs/overview#initializing-flutterfire). 

Unfortunately, this Flutter side configuration is not sufficient for a proper configuration of the firebase plugins. For Android and iOS it is still required to generate the google-services.json (or google-services.plist) files from the firebase console and copy them into the corresponding project folders. If you have for example 3 flavors in your application, then you need to add 3 google-services files for Android and 3 files for iOS. It would be helpful to pass the firebase configuration options via flutter code only (as it is already possible for flutter web) to avoid tedious native configuration. I see the following advantages of this approach:

1. avoid redundant configuration 
2. API keys are compiled into binary code and can be obfuscated. 
3. Easier possibility to provide the firebase configuration from external values (e. g. by using --dart-define build options)

**Comment by Ehesp** [comment]
Hello,

Just to give more context on this topic:

1. Removing the `google-services.json` & `Google-Info.plist` files is totally do-able.
2. The idea would be to pass in options to `initializeApp` much like the web SDK does.
3. Each platform (web, android & iOS) has their own properties which isn't exposed from the web configuration. To get these properties, the user would have to download each of the config files and manually pluck the data out.

The reason we opted to not do this, is that for newer users this is more complicated that just "download and add a file here". That being said, I think this is something we can do as more of an advanced workflow, however I still believe the existing way is best for the standard approach.

For those users more experienced, I agree this can be beneficial.

**Comment by devjeff** [comment]
What priority does this request have? What is required to plan the implementation of this request into an upcoming release?
